bundles/org.openhab.binding.huesync/src/main/java/org/openhab/binding/huesync/internal/connection/HueSyncConnection.java

```diff
@@ -95,59 +93,57 @@ public HueSyncConnection(HttpClient httpClient, String host, Integer port, Strin
         this.httpClient = httpClient;
     }
 
+    @SuppressWarnings("null")
```

Comment:
Why did you add @SuppressWarnings ?  It is asking the compiler to hide a potential null error..

Reply:
^
I am guessing that you did this to avoid potential null warnings when a class method is accessing a non final class field which might be nulled (on a separate thread by some other method) while the current method is running. Or??

In which case the solution is as follows..

```
@NonNullByDefault
public MyClass {
  private @Nullable String myField = null;
  private void myMethod() {
    String myLocal = myField;
    if (myLocal != null) {
      // do something with myLocal
    }
  }
}
```

Reply:
Thanks for following along and your comments. Highly appreciated!

The idea is that `this.responseReceived(response)` contains the `null` check as well - so that I do not need to repeat this in every method receiving a response. As there is still duplicated code there will be changes in the communication and exception exception handling and the problem will solve itslef. But the annotations sometimes make my life and the code more complex - unless I missed something here the check is wrong and there is a null check. Unfortunately the local build reports different errors than the pipeline - so I have problems to assure my commits are 👍. 

➡️ Is there a way to inform the analyzer that a method performs such a check - or do we need to add boilerplate code? I prefered the annotation - but unfortunately I can only do it on method level - not per line as far as I could find.

Reply:
🙂 - now I'm happy - it fails locally as well with the same errors 👍 - now I can work efficiently to resolve the warnings and errors. I did not execute a `mvn clean` - with the following sequence things are now consistent:

```bash
#!/usr/bin/bash

sudo truncate --size 0 /home/pgfeller/Temp/openhab/userdata/logs/*

mvn clean -f "/home/pgfeller/Documents/GitHub/openhab-addons/bundles/org.openhab.binding.huesync/pom.xml"
mvn spotless:apply
mvn i18n:generate-default-translations
mvn package -f /home/pgfeller/Documents/GitHub/openhab-addons/bundles/org.openhab.binding.huesync/pom.xml

cp --force /home/pgfeller/Documents/GitHub/openhab-addons/bundles/org.openhab.binding.huesync/target/org.openhab.binding.huesync-4.2.0-SNAPSHOT.jar \
           /home/pgfeller/Temp/openhab/addons/

docker container start openhab-development-server
```
➡️ I did hide my previous comment as resolved - and will now work on resolving the actual problems 😉.